service/controller/resource/azuremachinepoolconditions/conditionsubnetready.go

```diff
@@ -0,0 +1,111 @@
+package azuremachinepoolconditions
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-11-01/network"
+	azureconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions/azure"
+	"github.com/giantswarm/microerror"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+
+	"github.com/giantswarm/azure-operator/v5/pkg/helpers"
+)
+
+const (
+	subnetDeploymentPrefix     = "subnet"
+	provisioningStateSucceeded = "Succeeded"
+
+	SubnetNotFoundReason          = "SubnetNotFound"
+	SubnetProvisioningStatePrefix = "SubnetProvisioningState"
+)
+
+func (r *Resource) ensureSubnetReadyCondition(ctx context.Context, azureMachinePool *capzexp.AzureMachinePool) error {
+	r.logDebug(ctx, "ensuring condition %s", azureconditions.SubnetReadyCondition)
+
+	// Get Azure deployments client
+	deploymentsClient, err := r.azureClientsFactory.GetDeploymentsClient(ctx, azureMachinePool.ObjectMeta)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	// Now let's first check ARM deployment state
+	subnetDeploymentName := getSubnetDeploymentName(azureMachinePool.Name)
+	isSubnetDeploymentSuccessful, err := r.checkIfDeploymentIsSuccessful(ctx, deploymentsClient, azureMachinePool, subnetDeploymentName, azureconditions.SubnetReadyCondition)
+	if err != nil {
+		return microerror.Mask(err)
+	} else if !isSubnetDeploymentSuccessful {
+		// in the deployment is not yet successful, the check method will set
```

Comment:
Can we improve this comment? I'm not sure what's the "check method".

Reply:
this comment is not even proper English :smile: